granularflow.go

```diff
@@ -110,9 +111,15 @@ func (c *baseConn) pesimistic(r *buff.Reader, q *gfQuery) error {
 }
 
 func (c *baseConn) prepare(r *buff.Reader, q *gfQuery) (idPair, error) {
+	headers := copyHeaders(q.headers)
+
+	if c.protocolVersion.gte(version{0, 10}) {
```

Comment:
I think we should set some internal bool flag like `c.ver_0_10` to avoid this function call.